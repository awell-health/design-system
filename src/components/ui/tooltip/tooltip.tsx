import * as React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/utils';

const tooltipVariants = cva('tooltip before:shadow before:px-3 before:py-2 before:rounded-lg', {
  variants: {
    variant: {
      default: '',
      light: 'before:bg-white before:text-slate-700 after:'
    },
    placement: {
      top: 'tooltip-top',
      bottom: 'tooltip-bottom',
      left: 'tooltip-left',
      right: 'tooltip-right'
    }
  },
  defaultVariants: {
    variant: 'default'
  }
});

export interface TooltipProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof tooltipVariants> {
  children: JSX.Element | string;
  datatip: JSX.Element | string;
}

function Tooltip({ datatip, className, children, variant, placement, ...props }: TooltipProps) {
  return (
    <div
      data-tip={datatip}
      className={cn(tooltipVariants({ variant, placement }), className)}
      {...props}
    >
      {children}
    </div>
  );
}

export { Tooltip, tooltipVariants };
