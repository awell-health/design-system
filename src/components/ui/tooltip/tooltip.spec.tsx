import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import { Tooltip } from './tooltip';

describe('Tooltip', () => {
  it('match snapshot', () => {
    const result = render(
      <Tooltip datatip='Tooltip displayed text' data-testid='tooltip'>
        <span>Tooltip child</span>
      </Tooltip>
    );
    expect(result).toMatchSnapshot();
  });
});
