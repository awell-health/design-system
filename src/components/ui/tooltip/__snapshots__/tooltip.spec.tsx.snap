// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip > match snapshot 1`] = `
<div>
  <span
    data-tooltip-id="tooltip-id"
  >
    Tooltip child
  </span>
</div>
`;
